src/DotVVM.Framework/Compilation/ViewCompilingVisitor.cs

```diff
@@ -39,21 +42,25 @@ public override void VisitView(ResolvedTreeRoot view)
 
             var createsCustomDerivedType = view.Metadata.Type == typeof(DotvvmView);
 
+            string resultControlType;
             if (createsCustomDerivedType)
             {
-                emitter.ResultControlType = className + "Control";
-                emitter.EmitControlClass(view.Metadata.Type, emitter.ResultControlType);
+                resultControlType = className + "Control";
```

Comment:
In this case, `resultControlType` is always relative name, so we will end up using expression `SyntaxFactory.ParseTypeName(resultControlType)`.